textrepo-app/src/main/java/nl/knaw/huc/service/task/importer/JdbiImportFileTaskBuilder.java

```diff
@@ -127,16 +146,24 @@ public ResultImportDocument run() {
       final Contents contents = ResourceUtils.readContents(inputStream);
 
       // Now that 'contents' is ready, enter transaction to update document, file, version and contents
-      return jdbi.inTransaction(transaction -> {
+      var result = jdbi.inTransaction(transaction -> {
         final var doc = documentFinder.executeIn(transaction);
-        final var file = new HaveFileForDocumentByType(idGenerator, doc, typeName).executeIn(transaction);
+        var file = new HaveFileForDocumentByType(idGenerator, doc, typeName).executeIn(transaction);
         new SetFileProvenance(file, filename).executeIn(transaction);
         final var justBeforeCreation = now();
         final var version = new SetFileContents(idGenerator, file, contents, asLatestVersion)
             .executeIn(transaction);
         final var wasCreatedInThisRun = version.getCreatedAt().isAfter(justBeforeCreation);
-        return new ResultImportDocument(doc, version, wasCreatedInThisRun);
+        return new ResultImportDocument(doc, file, version, wasCreatedInThisRun);
       });
+
+      if (indexing) {
```

Comment:
Encapsuleren in een IndexFile() object?

Reply:
De enige relevante stap om te encapsuleren is het aanroepen van de indexerService: dat is waarschijnlijk te weinig